exp/addons/internal/controllers/clusterresourceset_helpers.go

```diff
@@ -108,27 +109,36 @@ func createUnstructured(ctx context.Context, c client.Client, obj *unstructured.
 
 // getOrCreateClusterResourceSetBinding retrieves ClusterResourceSetBinding resource owned by the cluster or create a new one if not found.
 func (r *ClusterResourceSetReconciler) getOrCreateClusterResourceSetBinding(ctx context.Context, cluster *clusterv1.Cluster, clusterResourceSet *addonsv1.ClusterResourceSet) (*addonsv1.ClusterResourceSetBinding, error) {
-	clusterResourceSetBinding := &addonsv1.ClusterResourceSetBinding{}
-	clusterResourceSetBindingKey := client.ObjectKey{
-		Namespace: cluster.Namespace,
-		Name:      cluster.Name,
+	clusterResourceSetBinding := &addonsv1.ClusterResourceSetBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cluster.Name,
+			Namespace: cluster.Namespace,
+		},
 	}
+	clusterResourceSetBindingKey := client.ObjectKeyFromObject(clusterResourceSetBinding)
 
 	if err := r.Client.Get(ctx, clusterResourceSetBindingKey, clusterResourceSetBinding); err != nil {
 		if !apierrors.IsNotFound(err) {
 			return nil, err
 		}
-		clusterResourceSetBinding.Name = cluster.Name
-		clusterResourceSetBinding.Namespace = cluster.Namespace
-		clusterResourceSetBinding.OwnerReferences = []metav1.OwnerReference{
-			{
-				APIVersion: addonsv1.GroupVersion.String(),
-				Kind:       "ClusterResourceSet",
-				Name:       clusterResourceSet.Name,
-				UID:        clusterResourceSet.UID,
+		err = controllerutil.SetOwnerReference(
+			&metav1.PartialObjectMetadata{
```

Comment:
q: Is there a reason why we are using a PartialObjectMetadata instead of using clusterResourceSet

Reply:
No specific reason, both work here.

Reply:
Switched to clusterResourceSet to avoid any confusion.